fix: exclude generated server.js from ESLint

## Description
The generated `server.js` file in `packages/vscode-pike/server/` is being linted by ESLint, producing 25+ errors. This file is auto-generated and should be excluded from linting.

## Problem
Running `bun run lint` produces errors like:
```
/home/smuks/OpenCode/pike-lsp/packages/vscode-pike/server/server.js
    1:440     error    Expected an assignment or function call and instead saw an expression
```
These errors are noise - the generated file is not meant to be linted directly.

## Expected Behavior
ESLint should exclude generated JavaScript files in the server output directory.

## Suggested Approach
Add `packages/vscode-pike/server/**/*.js` to the `ignores` array in `eslint.config.mjs`.

## Affected Files
- `eslint.config.mjs`: add ignore pattern for generated JS files

## Acceptance
Run `bun run lint` - no errors from server.js files.

## Research Done
- Searched closed issues: none found
- Reproduced: YES - lint shows 25+ errors for server.js

## Environment
- Pike binary: Pike v8.0 release 1116 Copyright © 1994-2020 Linköping University
- Bun version: 1.3.8

--- a/eslint.config.mjs
+++ b/eslint.config.mjs
@@ -18,6 +18,7 @@ export default [
             '**/.planning/**',
             '**/*.test.ts',
             'packages/vscode-pike/**/*.ts',
+            'packages/vscode-pike/server/**/*.js',
             'docs/**',
         ],
     },

